### PR TITLE
[ext_proc]: Refactor ext proc timeout handling

### PR DESCRIPTION
**Pull request by chaoqin-li1123**
Commit Message: Move setCallbackState and startTimer or disableTimer into the same function call because callback_state and timer have the same lifetime. This will make the state machine management more consistent and easier to reason about.
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/http/ext_proc/processor_state.cc [hunk 1]
```diff
@@ -18,13 +18,22 @@ using envoy::service::ext_proc::v3::CommonResponse;
 using envoy::service::ext_proc::v3::HeadersResponse;
 using envoy::service::ext_proc::v3::TrailersResponse;
 
-void ProcessorState::startMessageTimer(Event::TimerCb cb, std::chrono::milliseconds timeout) {
+void ProcessorState::onStartCall(Event::TimerCb cb, std::chrono::milliseconds timeout,
```
Comment (snowp) [inline source/extensions/filters/http/ext_proc/processor_state.cc:21]: Maybe name this (and the other function) super clear that we're talking about a call to the processor? e.g. `startProcessorCall`, `onStartProcessorCall` etc

**Review by snowp — CHANGES_REQUESTED** [review]
Nice, seems like an improvement. One possible suggestion, but general lgtm

File: source/extensions/filters/http/ext_proc/processor_state.cc (hunk 1 above)
Comment (chaoqin-li1123) [reply source/extensions/filters/http/ext_proc/processor_state.cc]: Thanks, I have renamed the method.

**Review by snowp — APPROVED** [review]
lgtm, thanks!

**Pull request merged**